plugins/outputs/mqtt/topic_name_generator_test.go

```diff
@@ -0,0 +1,76 @@
+package mqtt
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/plugins/serializers"
+	"github.com/influxdata/telegraf/testutil"
+)
+
+func Test_generateTopicName(t *testing.T) {
```

Comment:
You can also move that function into `mqtt_test.go`.

Reply:
Will do as you recommend, but I am confused for why do you have this preference. I would like to learn more about it. I was under the impression was better to keep the test files specific to what they are testing. This test function is testing only the `TopicNameGenerator`, so I thought to keep it in an specific file.

Thoughts?

Reply:
It's not a must, I said you _can_. :-) Distributing stuff too much, especially when files are short, will require people to navigate around, but if you prefer to keep it the way it is that's fine with me.